### ISSUE TITLE
Performance gap between repo results and paper – looking for tuning suggestions

### ISSUE THREAD

**Issue by TzJ2006** [HarryLui98/DMPEL#1]
Hi DMPEL team, thanks for releasing the code! I’ve managed to run the full lifelong-learning pipeline on the `libero_goal` benchmark, but the success rates I obtain are lower than those reported in the paper. I’d like to confirm whether I’m missing any important tricks or recommended hyper-parameter settings.

**Reproduction details**

Command: 
bash exp_scripts/lifelong_scripts/dmpel.sh:
```
torchrun --standalone --nproc_per_node=gpu libero/lifelong/main.py seed=100 benchmark_name=libero_goal \
        policy=bc_foundation_dmpel_policy lifelong=dmpel \
        exp=/work/DMPEL/checkpoints/lifelong/dmpel/goal \
        pretrain_model_path=/work/DMPEL/checkpoints/multitask_model_ep10.pth
```

GPU: 1 x A100
CUDA / Driver: CUDA12.0, Driver Version 525.147.05
python: 3.8.13
package: same as requirements.txt
seed: 100

**My Results**

| Method | Paper’s success rate (LIBERO-Goal, 10 tasks)                | Re-run success rate |
| ------ | ----------------------------------------------------------- | ------------------- |
| DMPEL  | *≈ 28 %* (Figure 5, “Lifelong” column)  | < 5 %         |
| ISCIL  | *≈ 23 %* (same figure)                                      | < 5 %           |

**Additional findings**
- Pre-trained checkpoint sanity-check
The provided multitask checkpoint matches the paper on its pre-training tasks (good success rate, most of them over 85%).

**Questions**
- Is there a specific environment configuration (e.g., in MuJoCo or Robosuite) that significantly impacts performance?
- Do specific seeds, task-order curriculum strategies, or “hidden” hyperparameters (beyond what’s documented) affect performance in a meaningful way?
- Could you advise on recommended approaches or tools for visualizing results (e.g., exporting the robot move to .mp4 files) to diagnose where the pipeline may be underperforming?

**Comment by HarryLui98** [comment]
Dear [TzJ2006](https://github.com/TzJ2006), 

Hi! Thank you very much for your interest in our work. I just re-checked and ran the code again, and I realized that I forgot to change n_epochs=1 to n_epochs=10 in the final uploaded code (`libero/configs/train/default.yaml`). You can fix this by specifying `train.n_epochs=10` in the `dmpel.sh`. Therefore, I guess the results you obtained previously were actually the zero-shot success rate of the pre-trained policy in `Goal`. 

All results reported in the paper are based on training for 10 epochs (with evaluation every 2 epochs). I have also updated the default config file so that the default number of epochs is now 10. If everything is set correctly, running `dmpel.sh` should output similar result as follows (`seed=300`): 

```
[info] # Batch: 220 | Epoch:   0 | train loss: -9.17 | time: 0.94 | Memory utilization: 5.262 GB
[info] evaluate task 0 takes 267.2 seconds with average inference time 53.2 ms and success rate 0.05
[info] Epoch:   0 | succ: 0.05 ± 0.10 | best succ: 0.05 | succ. AoC 0.05 | time: 4.45
[info] # Batch: 220 | Epoch:   1 | train loss: -15.08 | time: 1.60 | Memory utilization: 18.378 GB
[info] # Batch: 220 | Epoch:   2 | train loss: -16.91 | time: 1.54 | Memory utilization: 18.378 GB
[info] evaluate task 0 takes 265.5 seconds with average inference time 55.6 ms and success rate 0.65
[info] Epoch:   2 | succ: 0.65 ± 0.21 | best succ: 0.65 | succ. AoC 0.35 | time: 4.43
[info] # Batch: 220 | Epoch:   3 | train loss: -17.43 | time: 1.49 | Memory utilization: 18.378 GB
[info] # Batch: 220 | Epoch:   4 | train loss: -17.81 | time: 1.63 | Memory utilization: 18.378 GB
[info] evaluate task 0 takes 267.5 seconds with average inference time 53.8 ms and success rate 0.70
[info] Epoch:   4 | succ: 0.70 ± 0.20 | best succ: 0.7 | succ. AoC 0.47 | time: 4.46
[info] # Batch: 220 | Epoch:   5 | train loss: -18.03 | time: 1.52 | Memory utilization: 18.378 GB
[info] # Batch: 220 | Epoch:   6 | train loss: -18.30 | time: 1.46 | Memory utilization: 18.378 GB
[info] evaluate task 0 takes 257.4 seconds with average inference time 52.1 ms and success rate 0.80
[info] Epoch:   6 | succ: 0.80 ± 0.18 | best succ: 0.8 | succ. AoC 0.55 | time: 4.29
[info] # Batch: 220 | Epoch:   7 | train loss: -18.48 | time: 1.63 | Memory utilization: 18.378 GB
[info] # Batch: 220 | Epoch:   8 | train loss: -18.70 | time: 1.54 | Memory utilization: 18.378 GB
[info] evaluate task 0 takes 241.8 seconds with average inference time 47.9 ms and success rate 0.75
[info] Epoch:   8 | succ: 0.75 ± 0.18 | best succ: 0.8 | succ. AoC 0.55 | time: 4.03
[info] # Batch: 220 | Epoch:   9 | train loss: -18.83 | time: 1.50 | Memory utilization: 18.378 GB
[info] # Batch: 220 | Epoch:  10 | train loss: -18.99 | time: 1.59 | Memory utilization: 18.378 GB
[info] evaluate task 0 takes 267.9 seconds with average inference time 56.9 ms and success rate 0.85
[info] Epoch:  10 | succ: 0.85 ± 0.16 | best succ: 0.85 | succ. AoC 0.76 | time: 4.47

```

Additionally, we use the default settings and task order provided in LIBERO, as well as the default environment configurations in Robosuite/MuJoCo. Our experiments with three random seeds indicate that there are no significant discrepancies in average performance across different seeds (as shown in Figure 4 in the paper).

Best regards,
Yuheng

**Comment by Holmes-GU** [comment]
> Hi, I rerun the experiment with the corrected setting (train.n_epochs=10) and obtained great results. Thanks again for advising me on the configuration fix—I really appreciate your help!

Hi, would you like to share how you run the full codes, I have encountered some issues. Thank you very much
